### PR TITLE
Small improvements to tox configuration

### PR DESCRIPTION
**Pull request by RobbeSneyders**
Had issues running tox locally which in the end were unrelated to the configuration. But in the meantime I learned some better practices for tox, which I implemented in this PR.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -13,33 +13,41 @@ python =
...
+commands_pre=
     poetry lock
     poetry install --all-extras
     poetry show
     poetry export -f requirements.txt --output /tmp/requirements.txt
+commands=
     poetry run liccheck -s license_strategy.ini -r /tmp/requirements.txt -l PARANOID
 
-[testenv]
+[testenv:py{38,39,310,311}]
```
Comment (RobbeSneyders) [inline tox.ini:35]: No longer inherit this configuration in the other test environments, which lead to unnecessary dependency installation etc.

File: tox.ini [hunk 2]
```diff
@@ -13,33 +13,41 @@ python =
...
     /usr/bin/bash
-commands=
+commands_pre=
     bash ./scripts/pre-build.sh
     poetry lock
     poetry install --all-extras
     poetry show
+commands=
     poetry run python -m pytest tests -vv --cov fondant --cov-report term-missing
+commands_post=
```
Comment (RobbeSneyders) [inline tox.ini:52]: Split pre and post commands so post script always run, even when command fails.

File: tox.ini [hunk 3]
```diff
@@ -13,33 +13,41 @@ python =
...
 allowlist_externals=
     pre-commit
 commands=pre-commit run --all-files --show-diff-on-failure
 
 [testenv:check-licenses]
+skip_install=true
 deps=
     liccheck>=0.7.3,<0.8
-commands=
+allowlist_externals=
```
Comment (RobbeSneyders) [inline tox.ini:25]: No longer inherited

File: tox.ini [hunk 4]
```diff
@@ -13,33 +13,41 @@ python =
     3.11: py311
 
 [testenv:pre-commit]
+skip_install=true
```
Comment (RobbeSneyders) [inline tox.ini:16]: Not necessary to install fondant.

**Review by mrchtr — APPROVED** [review]
Thanks @RobbeSneyders! I have tested your changes locally, and everything is working.

**Comment by GeorgesLorre** [comment]
Should be faster as well no ? With less installing of packages.

**Comment by RobbeSneyders** [comment]
> Should be faster as well no ? With less installing of packages.

Yes, running a single pytest environment takes ~20 seconds for me now.

**Pull request merged**